### PR TITLE
Add export feature

### LINKED ISSUE #14: As a Hall admin general, i can export a list of emails
so that I can mass email the correct group of residents

### PR DESCRIPTION
**Pull request by yanchenglee98**
Add export feature.
Close #14

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ExportCommand.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * Creates an ExportCommand to handle the specified information
+     */
+    public ExportCommand(String type) {
+        extractType = type;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+
```
Comment (teekoksiang) [inline src/main/java/seedu/address/logic/commands/ExportCommand.java:35]: Omit the blank line?

File: src/main/java/seedu/address/logic/parser/ExportCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import seedu.address.logic.commands.ExportCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class ExportCommandParser implements Parser<ExportCommand> {
+
+    public static final String INVALID_EXPORT_COMMAND = "Invalid command format!\n"
+            + "export: Exports the corresponding information to a text file\n"
+            + "Parameters: INFORMATION (e.g. email, phone)\n"
+            + "Example: export email";
```
Comment (teekoksiang) [inline src/main/java/seedu/address/logic/parser/ExportCommandParser.java]: This part perhaps can declare it as `MESSAGE_USAGE` in the `ExportCommand`?
So, it follows the current exception handling approach. 
Can refer to AddCommand and AddCommandParser.

File: src/main/java/seedu/address/logic/parser/ExportCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    /**
+     * Parses the given {@code String} of arguments in the context of the ExportCommand
+     * and returns an ExportCommand object for execution.
+     * @throws ParseException if the user input does not conform to the expected format
+     */
+    public ExportCommand parse(String args) throws ParseException {
+        String[] split = args.split(" ");
+
+        if (split.length < 2) {
```
Comment (teekoksiang) [inline src/main/java/seedu/address/logic/parser/ExportCommandParser.java]: Perhaps can replace `1` and `2` with named constant to improve the readability?

File: src/main/java/seedu/address/logic/commands/ExportCommand.java (hunk 1 above)
Comment (teekoksiang) [inline src/main/java/seedu/address/logic/commands/ExportCommand.java]: Omit this blank line?

File: src/main/java/seedu/address/logic/commands/ExportCommand.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        File directory = new File(DIRECTORY_NAME);
+
+        // checks if directory exists, else directory will be created
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        try {
+            FileWriter fw = new FileWriter(FILENAME);
+            List<Person> list = model.getAddressBook().getPersonList();
```
Comment (teekoksiang) [inline src/main/java/seedu/address/logic/commands/ExportCommand.java]: `handleEmail` and `handlePhone` both need to create the file and get the person list first.
 Can consider declaring a method to do this part, then pass the person list to `handleEmail` and `handlePhone`.

**Review by teekoksiang — COMMENTED** [review]
Almost LGTM, some minor things to consider.

File: src/main/java/seedu/address/logic/commands/ExportCommand.java (hunk 1 above)
Comment (yanchenglee98) [reply src/main/java/seedu/address/logic/commands/ExportCommand.java]: Ok thanks

File: src/main/java/seedu/address/logic/commands/ExportCommand.java (hunk 4 above)
Comment (yanchenglee98) [reply src/main/java/seedu/address/logic/commands/ExportCommand.java]: Ok sure but is there a difference between passing the list to the method or getting the person list in the method?

File: src/main/java/seedu/address/logic/parser/ExportCommandParser.java (hunk 2 above)
Comment (yanchenglee98) [reply src/main/java/seedu/address/logic/parser/ExportCommandParser.java]: Will do thanks

File: src/main/java/seedu/address/logic/parser/ExportCommandParser.java (hunk 3 above)
Comment (yanchenglee98) [reply src/main/java/seedu/address/logic/parser/ExportCommandParser.java]: Will do thanks

File: src/main/java/seedu/address/logic/commands/ExportCommand.java (hunk 4 above)
Comment (teekoksiang) [reply src/main/java/seedu/address/logic/commands/ExportCommand.java]: The difference is minor, mainly to reduce the repeated codes.

**Review by teekoksiang — APPROVED** [review]
LGTM, merging.

**Pull request merged**